client/blocks/edit-gravatar/index.jsx

```diff
@@ -153,11 +154,56 @@ export class EditGravatar extends Component {
 	};
 
 	render() {
-		const { isUploading, translate, user, additionalUploadHtml } = this.props;
+		const { isUploading, translate, user, additionalUploadHtml, userSettings } = this.props;
 		const gravatarLink = `https://gravatar.com/${ user.username || '' }`;
 		// use imgSize = 400 for caching
 		// it's the popular value for large Gravatars in Calypso
 		const GRAVATAR_IMG_SIZE = 400;
+
+		if ( userSettings.gravatar_profile_hidden ) {
```

Comment:
It's too bad that this hidden profile version has to involve so much copy-paste, but I think I can see how it's _just_ different enough from the public profile version that it'd be just as messy to try and merge the two cases together but using if/elses to change what's needed. 😮‍💨

Reply:
In the future, we'll be able to write code with our minds, and we won't even have to worry about which file the code goes in!